src/shaders/_prelude.fragment.glsl

```diff
@@ -31,11 +31,11 @@ vec3 dither(vec3 color, highp vec2 seed) {
 
 // Pack depth to RGBA. A piece of code copied in various libraries and WebGL
 // shadow mapping examples.
-vec4 pack_depth(float ndc_z) {
-    float depth = ndc_z * 0.5 + 0.5;
-    const vec4 bit_shift = vec4(256.0 * 256.0 * 256.0, 256.0 * 256.0, 256.0, 1.0);
-    const vec4 bit_mask  = vec4(0.0, 1.0 / 256.0, 1.0 / 256.0, 1.0 / 256.0);
-    vec4 res = fract(depth * bit_shift);
+highp vec4 pack_depth(highp float ndc_z) {
```

Comment:
Is it necessary to set default precision to `mediump` if it's overwritten to be `highp` for most of the variables? Default precision of floats should be `highp` (glsl 1.00.17) so the initial statement could be omitted.

Reply:
> Do you know what was the cause of this regression? Could you also add a regression test in order to catch these cases in the future?

It was caused because I moved this function from globe_depth and terrain_depth to the fragment prelude and forgot to check the precision. I think we can't test it here because it needs GLES, but the test cases in gl-native found it.

> Is it necessary to set default precision to mediump if it's overwritten to be highp for most of the variables? Default precision of floats should be highp (glsl 1.00.17) so the initial statement could be omitted.

I thought about that, but the problem is that this prelude is included in all fragment shaders and if I change the precision to highp here it breaks some things in those other shaders, so the default is not only used in this context.

Reply:
that's a good point! :+1: